timelock-corruption-detection/src/main/java/com/palantir/timelock/history/remote/HistoryLoaderAndTransformer.java

```diff
@@ -36,8 +38,11 @@ private HistoryLoaderAndTransformer() {
 
     public static List<LogsForNamespaceAndUseCase> getLogsForHistoryQueries(
             LocalHistoryLoader localHistoryLoader, List<HistoryQuery> historyQueries) {
-        Map<NamespaceAndUseCase, Long> lastVerifiedSequences = historyQueries.stream()
-                .collect(Collectors.toMap(HistoryQuery::getNamespaceAndUseCase, HistoryQuery::getSeq, Math::min));
+        Map<NamespaceAndUseCase, SequenceBounds> lastVerifiedSequences = historyQueries.stream()
+                .collect(Collectors.toMap(
```

Comment:
I'm not a huge fan - `KeyedStream.of(historyQueries).mapKeys(...).map(...)` gets you most of the way, although I'll confess that I'm not sure how to do the merging resolution in `KeyedStream` (although you can always collect to multimap).

Reply:
I think what we have now is more readable